ocaml/otherlibs/systhreads/st_pthreads.h

```diff
@@ -204,8 +204,8 @@ Caml_inline void st_thread_yield(st_masterlock * m)
   }
 
   m->busy = 0;
-  atomic_fetch_add(&m->waiters, +1);
   custom_condvar_signal(&m->is_free);
+  atomic_fetch_add(&m->waiters, +1);
```

Comment:
This change looks bad: in a 2-thread case it means that there's _more_ work between the owning thread signalling and waiting, _increasing_ the chances that the woken thread immediately sleeps again to wait for the lock.